# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 786805a2cdcc
# Full Hash: 786805a2cdcc4cdab632a955dace417062d0087e
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-11-30 04:12:04
# Regressor Bug: 1862802
# File Overlap Count: 1
# Description:
#   Bug 1862802 part 5: Remove virtual cursor change event. r=eeejay
#   
#   Depends on D192645
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192646
# ==============================================================================

diff -r effe12f2120a -r 786805a2cdcc accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Tue Nov 28 22:03:20 2023 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Tue Nov 28 22:03:21 2023 +0000
@@ -936,16 +936,6 @@
                                      aEvent->GetEventType());
           break;
         }
-        case nsIAccessibleEvent::EVENT_VIRTUALCURSOR_CHANGED: {
-          AccVCChangeEvent* vcEvent = downcast_accEvent(aEvent);
-          LocalAccessible* position = vcEvent->NewAccessible();
-          LocalAccessible* oldPosition = vcEvent->OldAccessible();
-          ipcDoc->SendVirtualCursorChangeEvent(
-              id, oldPosition ? oldPosition->ID() : 0,
-              position ? position->ID() : 0, vcEvent->Reason(),
-              vcEvent->IsFromUserInput());
-          break;
-        }
         case nsIAccessibleEvent::EVENT_FOCUS:
           ipcDoc->SendFocusEvent(id);
           break;
@@ -1061,15 +1051,6 @@
                              aEvent->GetEventType());
       break;
     }
-    case nsIAccessibleEvent::EVENT_VIRTUALCURSOR_CHANGED: {
-#ifdef ANDROID
-      AccVCChangeEvent* vcEvent = downcast_accEvent(aEvent);
-      PlatformVirtualCursorChangeEvent(
-          target, vcEvent->OldAccessible(), vcEvent->NewAccessible(),
-          vcEvent->Reason(), vcEvent->IsFromUserInput());
-#endif
-      break;
-    }
     case nsIAccessibleEvent::EVENT_FOCUS: {
       LayoutDeviceIntRect rect;
       // The caret rect is only used on Windows, so just pass an empty rect on